langchain_tools_demo/orchestration/orchestration.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Generic, List, TypeVar
+
+import aiohttp
+
+# aiohttp context
+connector = None
+
+
+class classproperty:
+    def __init__(self, func):
+        self.fget = func
+
+    def __get__(self, instance, owner):
+        return self.fget(owner)
+
+
+class Orchestration(ABC):
+    client: aiohttp.ClientSession
+
+    @classproperty
+    @abstractmethod
+    def kind(cls):
+        pass
+
+    @classmethod
+    @abstractmethod
+    async def create(cls, history: List[Any]) -> "Orchestration":
+        pass
+
+    async def invoke(self, prompt: str):
+        raise NotImplementedError("Subclass should implement this!")
+
+    @staticmethod
+    async def get_connector():
+        global connector
+        if connector is None:
+            connector = aiohttp.TCPConnector(limit=100)
+        return connector
+
+    @staticmethod
+    async def handle_error_response(response):
+        if response.status != 200:
+            return f"Error sending {response.method} request to {str(response.url)}): {await response.text()}"
```

Comment:
If we want a static method we can just put it as a utility somewhere -- we don't need to nest on the object

Reply:
Removed this function instead. Seems like we no longer use this.